include/envoy/http/header_map.h

```diff
@@ -774,6 +776,15 @@ class HeaderMatcher {
 public:
   virtual ~HeaderMatcher() = default;
 
+  /**
+   * This getter exists so that once matchHeaders() has been called on a HeaderMap the concrete
+   * header in that map can be extracted by using this getter to discover its name. This is possible
+   * in all cases, except when invert_match_ is true, because in the latter case name_ would *not*
+   * reflect the matching header.
+   * @return the header name the matcher will match against.
+   */
+  virtual const LowerCaseString* name() const PURE;
```

Comment:
I'm confused how this can be idempotent. If you need this information shouldn't it be returned by `matchesHeaders`?

Reply:
So the way the HeaderMatcher is used is first we call matchesHeaders(request_headers). If that returns true we still don't know which request header actually matched, only that one of them did. Thus I needed to add this getter function to extract what the header name actually is. With that I can get the value from request_headers.